org.eclipse.scanning.malcolm.core/src/org/eclipse/scanning/malcolm/core/AbstractMalcolmDevice.java

```diff
@@ -108,7 +114,7 @@ public void start(final IPosition pos) throws ScanningException, InterruptedExce
 			public void run() {
 				try {
 					AbstractMalcolmDevice.this.run(pos);
-				} catch (ScanningException|InterruptedException e) {
+				} catch (Exception e) {
 					e.printStackTrace();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)